File: redirections/ruby/redirections-Jeremy.rb
#!/usr/bin/env ruby
#
# Introduction to Redirections in ruby

# Error handling is a way of redirecting errors
begin
  sum = 6/0
rescue ZeroDivisionError
  puts 'Dividing by zero... very dangerous.'
else
  puts "Your sum is:n #{sum}"
ensure
  puts "And this gets executed anyway."
end

# Also can up $stdin and $stderr
$stdout.reopen('1.txt','a')
puts 'This will be put into 1.txt'

puts 'I can also put another line since its set to append'

$stdout.reopen('/dev/null','w')
puts 'Something in but nothing out!'

File: redirections/python/redirections-Jeremy.py
#!/usr/bin/env python
#
# Introduction to Redirections in python

# You need to import sys for basic redirects in python.
import sys

# The open function will use the current file the is open
# in this case I will be appending a file based on the second
# arguement of open()

# w -> write, r -> read, a -> append
sys.stdout = open('1.txt', 'a')
print 'This sentence will be put into a file named 1.txt'

print 'Since it is an append I can add a second line to the file'

# Switch the output of the print from the txt file to the
# /dev/null that deletes everything that comes in contact
# with it
sys.stdout = open('/dev/null', 'w')
print 'Welcome to the blackhole were data never comes out'

# Try/except/finally redirects the errors to do something else
# and not cause a fatal error that stops the script
try:
  import somerandomimportyoudonthave
except ImportError:
  somerandomimportyoudonthave = None
else:
  # Do the statement you wanted
  pass
finally:
  # This always gets executed
  pass
